lib/cache.js

```diff
@@ -8,8 +8,26 @@ const VERSION = require('../package').version;
 
 const TimeoutError = bluebird.TimeoutError;
 
-function createCacheKey(segment, id) {
+function createCacheKey(segment, ctx, opts) {
   const versionedSegment = `http-transport:${VERSION}:${segment}`;
+  const requiredIdPart = `${ctx.req.getMethod()}:${ctx.req.getUrl()}`;
+  let id = requiredIdPart;
```

Comment:
Potential for removing the mutable of `id` but returning with the default (`requiredIdPart`) ID if `varyOn` is false. 

Something like:
```js
  if (!varyOn) {
    return {
      segment: versionedSegment,
      id: requiredIdPart
    };

   // do vary on logic here 

    return {
       segment: versionedSegment,
       id: `${requiredIdPart}:${optionalIdPart.join(',')}`
  };

  }
```